Echo.Core/src/InOut/PolygonFileFormatReader.cs

```diff
@@ -133,11 +161,13 @@ float[] ReadBinaryFloats(int amount)
 		return floats;
 	}
 
-	uint[] ReadBinaryInts(int amount)
+	void ReadBinaryInts(int amount)
 	{
-		uint[] ints = new uint[amount];
-		for (int i = 0; i < amount; i++) ints[i] = ReadBinaryUInt();
-		return ints;
+		if (readUintBuffer.Length < amount)
+			readUintBuffer = new uint[amount];
+		//for (int i = 0; i < amount; i++) readUintBuffer[i] = ReadBinaryUInt();
+		file.Read(buffer, 0, amount * sizeof(uint));
+		Buffer.BlockCopy(buffer, 0, readUintBuffer, 0, amount * sizeof(uint));
 	}
 
 	public struct Triangle
```

Comment:
We can mark this `struct` as a `readonly struct`. It will improve readability and occasionally performance as well. Note that this will require us to make all of the fields `readonly`, meaning that a constructor must be introduced.